core/src/main/java/de/fraunhofer/ids/messaging/core/config/ClientProvider.java

```diff
@@ -180,8 +180,17 @@ private static void setProxySelector(final OkHttpClient.Builder okHttpBuilder,
                     if (log.isDebugEnabled()) {
                         log.debug("Address: [host=({})], Port: [port=({})]", proxyHost, proxyPort);
                     }
-                    proxyList.add(new Proxy(Proxy.Type.HTTP,
-                        new InetSocketAddress(proxyHost, proxyPort)));
+
+                    if (proxyHost == null || proxyHost.trim().equals("")) {
```

Comment:
Java URI.getPort() specifies:

```
 Returns: The port component of this URI, or -1 if the port is undefined
```

So maybe a check for unspecified ports should be added here, too.